### PR TITLE
docs: add readme for common and query

### LINKED ISSUE #7004: Add README.md in `common` to introduce the function of crates
**Summary**

We can add a `common/README.md` to introduce the function of common crates.

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

add readme for common and query

Fixes #7004 
Related #7050

### CODE REVIEW

File: src/common/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Databend Common
+
+- `arrow`, a simple wrapper for reading and writing parquet.
```
Comment (Xuanwo) [inline src/common/README.md:3]: How about providing a link?

File: src/query/README.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+- `ast`, new version of sql parser based on `nom-rule`.
+- `catalog` contains structures and traits for catalogs management, `Catalog`, `Database`, `Table` and `TableContext`.
+- `codegen` is used to generate the arithmetic result type.
+- `config` provides config support for databend query.
+- `datablocks`, `Vec` collection, which encapsulates some common methods, and will be gradually migrated to `expressions`.
+- `datavalues`, the definition of each type of Column, which represents the layout of data in memory, will be gradually migrated to `expressions`.
+- `expression`, the new scalar expression framework with expression definition (AST), type checking, and evaluation runtime.
+- `formats`, the serialization and deserialization of data in various formats to the outside.
+- `functions`, scalar functions and aggregate functions, etc.
```
Comment (Xuanwo) [inline src/query/README.md:13]: Please clarify this mod is going to be deprecated by `functions-v2`

File: src/query/README.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+- `functions-v2`, scalar functions and aggregate functions, etc., based on `expression`.
+- `legacy-parser`, the old parser, which will be replaced by ast, built with sqlparser-rs.
+- `menagement` for clusters, quotas, etc.
+- `pipeline` implements the scheduling framework for physical operators.
+- `planners` builds an execution plan from the user's SQL statement and represents the query with different types of relational operators.
+- `service` -> `databend-query`, the query service library of Databend.
```
Comment (Xuanwo) [inline src/query/README.md:19]: This comment is not related to this PR.

During my daily usage, I found that `service` is not very easy to find. I have to seek some time to get the correct location, and the `service` contains too many modules.

Any plan to address this?

File: src/common/README.md (hunk 1 above)
Comment (PsiACE) [reply src/common/README.md:3]: `[arrow](./arrow)` ?

File: src/query/README.md (hunk 3 above)
Comment (PsiACE) [reply src/query/README.md:19]: > service contains too many modules

Yes, so we still need to improve on this by making further differentiated definitions and providing better naming ...

There is no definite plan, but at least, as previously discussed in #6747, it is possible to at least split some crates from it.

**Review by Xuanwo — APPROVED** [review]
